subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,219 @@
+			ControlPlaneAgents: 3,
+			WorkerAgents:       0,
+			AgentSelector:      metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+		},
+	}
+}
+
+func compareSpecToCluster(cluster *common.Cluster, spec *v1alpha1.ClusterSpec) {
+	Expect(cluster.Name).Should(Equal(spec.Name))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.BaseDNSDomain).Should(Equal(spec.BaseDNSDomain))
+	Expect(cluster.ClusterNetworkCidr).Should(Equal(spec.ClusterNetworkCidr))
+	Expect(cluster.ClusterNetworkHostPrefix).Should(Equal(spec.ClusterNetworkHostPrefix))
+	Expect(cluster.ServiceNetworkCidr).Should(Equal(spec.ServiceNetworkCidr))
+	Expect(cluster.IngressVip).Should(Equal(spec.IngressVip))
+	Expect(cluster.SSHPublicKey).Should(Equal(spec.SSHPublicKey))
+	Expect(swag.BoolValue(cluster.VipDhcpAllocation)).Should(Equal(spec.VIPDhcpAllocation))
+	Expect(cluster.HTTPProxy).Should(Equal(spec.HTTPProxy))
+	Expect(cluster.HTTPSProxy).Should(Equal(spec.HTTPSProxy))
+	Expect(cluster.NoProxy).Should(Equal(spec.NoProxy))
+	Expect(swag.BoolValue(cluster.UserManagedNetworking)).Should(Equal(spec.UserManagedNetworking))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.AdditionalNtpSource).Should(Equal(spec.AdditionalNtpSource))
+	Expect(cluster.PullSecret).Should(Equal(pullSecret))
+}
+
+func cleanUP(ctx context.Context, client k8sclient.Client) {
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Image{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
+}
+
+var _ = Describe("cluster installation", func() {
+	if !Options.EnableKubeAPI {
+		return
+	}
+
+	ctx := context.Background()
+
+	waitForClusterReconcileTimeout := 10
+
+	secretRef := &corev1.SecretReference{}
+
+	generateFAPostStepReply := func(h *models.Host, freeAddresses models.FreeNetworksAddresses) {
+		fa, err := json.Marshal(&freeAddresses)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = agentBMClient.Installer.PostStepReply(ctx, &installer.PostStepReplyParams{
+			ClusterID: h.ClusterID,
+			HostID:    *h.ID,
+			Reply: &models.StepReply{
+				ExitCode: 0,
+				Output:   string(fa),
+				StepID:   string(models.StepTypeFreeNetworkAddresses),
+				StepType: models.StepTypeFreeNetworkAddresses,
+			},
+		})
+		Expect(err).To(BeNil())
+	}
+
+	secretRef = &corev1.SecretReference{
+		Namespace: Options.Namespace,
+		Name:      "pull-secret",
+	}
+	err := kubeClient.Get(ctx, types.NamespacedName{Namespace: Options.Namespace, Name: "pull-secret"}, &corev1.Secret{})
+	if apierrors.IsNotFound(err) {
+		deployPullSecretResource(ctx, kubeClient, secretRef.Name, pullSecret)
+	} else {
+		Expect(err).To(BeNil())
+	}
+
+	AfterEach(func() {
+		cleanUP(ctx, kubeClient)
+		clearDB()
+	})
+
+	It("[kube-api]deploy cluster with hosts and wait for preparing-for-installation", func() {
```

Comment:
i think that you can put it only on `Describe` because otherwice maybe `err := kubeClient.Get(ctx, types.NamespacedName{Namespace: Options.Namespace, Name: "pull-secret"}, &corev1.Secret{})`  will be called with none kube-api tests?